### PR TITLE
Add sarb logging option

### PR DESCRIPTION
**Pull request by BackEndTea**
This PR:

- [x] Adds the option to log a json file readable by [sarb](https://github.com/DaveLiddament/sarb), allowing a baseline feature

@DaveLiddament, could you look if the output is correct?

@theofidry Im gonna need some help with the failing tests, as i'm not sure how they work

### CODE REVIEW

File: src/Logger/SarbLogger.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+use function Safe\json_encode;
+
+/**
+ * @internal
+ */
+final class SarbLogger extends FileLogger
+{
+    protected function getLogLines(): array
+    {
+        $processes = array_merge(
```
Comment (theofidry) [inline src/Logger/SarbLogger.php]: missing `array_merge` in the importer symbols

File: src/Logger/SarbLogger.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            $this->metricsCalculator->getEscapedMutantProcesses(),
+            $this->metricsCalculator->getNotCoveredMutantProcesses()
+        );
+
+        $this->sortProcesses($processes);
+
+        $output = array_map(static function (MutantProcess $process): array {
```
Comment (theofidry) [inline src/Logger/SarbLogger.php]: `array_map` should be imported as well

File: tests/phpunit/Logger/SarbLoggerTest.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        $this->outputMock = $this->createMock(OutputInterface::class);
+    }
+
+    /**
+     * @dataProvider completeMetricsProvider
+     */
+    public function test_it_logs_results_in_a_text_file_when_there_are_mutations(
+        string $expectedContent
+    ): void {
+        $expectedContent = str_replace('', PHP_EOL, $expectedContent);
```
Comment (theofidry) [inline tests/phpunit/Logger/SarbLoggerTest.php]: `str_replace` and `PHP_EOL` should be imported

**Comment by theofidry** [comment]
@BackEndTea sure happy to help, feel free to ping me on Slack

File: tests/phpunit/Logger/SarbLoggerTest.php (hunk 3 above)
Comment (maks-rafalko) [reply tests/phpunit/Logger/SarbLoggerTest.php]: is it possible to automate this checks?

**Comment by theofidry** [comment]
@BackEndTea I'll rebase your PR myself once #1099 is merged

**Comment by theofidry** [comment]
Closed by #1114

**Pull request closed without merging**